src/Microsoft.DotNet.Interactive/KernelExtensions.cs

```diff
@@ -45,6 +45,29 @@ public static Kernel FindKernel(this Kernel kernel, string name)
             };
         }
 
+        public static Kernel FindKernel(this Kernel kernel, Func<Kernel, bool> selector)
```

Comment:
This should return `IEnumerable<Kernel>` since there can be more than one that matches.

It would be more straightforward to use `VisitSubkernels` and avoid looking at the `ChooseKernelDirective`s.